lib/datadog/tracing/contrib/aws/instrumentation.rb

```diff
@@ -60,6 +64,98 @@ def annotate!(span, context)
             span.set_tag(Tracing::Metadata::Ext::HTTP::TAG_STATUS_CODE, context.safely(:status_code))
           end
 
+          def add_service_specific_tags(span, aws_service, params)
+            case aws_service
+            when 'sqs'
+              add_sqs_tags(span, params)
+            when 'sns'
+              add_sns_tags(span, params)
+            when 'dynamodb'
+              add_dynamodb_tags(span, params)
+            when 'kinesis'
+              add_kinesis_tags(span, params)
+            when 'eventbridge'
+              add_eventbridge_tags(span, params)
+            when 'states'
+              add_states_tags(span, params)
+            when 's3'
+              add_s3_tags(span, params)
+            end
+          end
+
+          def add_sqs_tags(span, params)
+            queue_url = params[:queue_url]
+            queue_name = params[:queue_name]
+            # Regular expression to match the SQS queue URL pattern
+            # https://sqs.sa-east-1.amazonaws.com/12345678/MyQueueName
+            pattern = %r{https://sqs\.[^/]+\.amazonaws\.com/(\d+)/([^/]+)}
+
+            if queue_url && (match = queue_url.match(pattern))
+              aws_account, queue_name = match.captures
```

Comment:
👋 @zARODz11z, it seems like regular expression matching is quite a bit slower String#split.

You can run this benchmark locally by dropping this code snippet into a file and running `ruby that_file.rb`:

```ruby
# Run `gem install benchmark-ips` before running this file.
require 'benchmark/ips'

queue_url = "https://sqs.us-stubbed-1.amazonaws.com/123456789012/MyQueueName"

Benchmark.ips do |x|
  x.time = 5
  x.warmup = 0.2
  
  x.report('regex') do
    if queue_url && (match = queue_url.match(%r{https://sqs\.[^/]+\.amazonaws\.com/(\d+)/([^/]+)}))
      aws_account, queue_name = match.captures
    end
  end
  x.report('split') do
    if queue_url
      _,_,_, aws_account, queue_name = queue_url.split('/')
    end
  end

  x.compare!
end
```

Result:
```
# ruby 3.2.1
Comparison:
               split:  1915116.4 i/s
               regex:  1110268.4 i/s - 1.72x  slower
```